### PR TITLE
pyomo.network / pyomo.gdp interoperability

### PR DESCRIPTION
**Pull request by bknueven**
## Summary/Motivation:
This PR makes a few patches to pyomo.network and pyomo.gdp to better enable their combined use on the same Pyomo model. In particular, these changes allow Arcs to be placed on Disjuncts in a more seamless fashion.

## Changes proposed in this PR:
- Changes the behavior of the pyomo.network sequential decomposition to not descend into inactive Pyomo Blocks. As-is, pyomo.network Arcs on Disjuncts can cause issues with the sequential decomposition tool, even after using the gdp.fix_disjuncts transformation
- pyomo.gdp transformations now ignore active Ports (but will still complain about active Arcs if the network.expand_arcs transformation has not been called)
- network.expand_arcs will now descend into active Disjuncts by default, so that Arcs on Disjuncts get expanded

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/network/decomposition.py [hunk 1]
```diff
@@ -704,17 +706,18 @@ def create_graph(self, model):
...
-                                 "a graph for a model. Found undirected "
-                                 "Arc: '%s'" % arc.name)
-            if arc.expanded_block is None:
-                raise ValueError("All Arcs must be expanded when creating "
-                                 "a graph for a model. Found unexpanded "
-                                 "Arc: '%s'" % arc.name)
-            src, dest = arc.src.parent_block(), arc.dest.parent_block()
-            G.add_edge(src, dest, arc=arc)
+        for blk in model.block_data_objects(descend_into=True, active=True):
+            for arc in blk.component_data_objects(Arc, descend_into=False):
```
Comment (michaelbynum) [inline pyomo/network/decomposition.py:710]: Should we also ensure the arc is active?

**Review by michaelbynum — COMMENTED** [review]
Overall, this looks great. I do have one question below.

**Comment by michaelbynum** [comment]
Sorry - I did not notice that this was a draft PR.

**Comment by michaelbynum** [comment]
The failing tests seem unrelated by the way.

**Comment by bknueven** [comment]
> Sorry - I did not notice that this was a draft PR.

No, thanks. I wanted to get some feedback. I think some additional tests should be written to check the interoperability (hence "draft"), but I also didn't want to go through the trouble without some initial feedback, especially since this slightly alters the behavior of the network and gdp transformations, as well as the sequential decomposition algorithm.

File: pyomo/network/decomposition.py (hunk 1 above)
Comment (michaelbynum) [reply pyomo/network/decomposition.py:710]: Ahh, I see... Interesting. I need to process this a bit.

**Review by michaelbynum — APPROVED** [review]

**Review by michaelbynum — APPROVED** [review]
Looks great.

**Comment by bknueven** [comment]
I'm happy to resolve the conflicts, although I'm guessing #1960 will create more, so I'll hold off unless for some reason this should/will be merged first.

**Review by jsiirola — APPROVED** [review]

**Pull request merged**